sklearn/metrics/_ranking.py

```diff
@@ -1703,15 +1703,20 @@ def top_k_accuracy_score(
     y_type = type_of_target(y_true, input_name="y_true")
     if y_type == "binary" and labels is not None and len(labels) > 2:
         y_type = "multiclass"
-    y_score = check_array(y_score, ensure_2d=False)
-    y_score = column_or_1d(y_score) if y_type == "binary" else y_score
-    check_consistent_length(y_true, y_score, sample_weight)
-
     if y_type not in {"binary", "multiclass"}:
         raise ValueError(
             f"y type must be 'binary' or 'multiclass', got '{y_type}' instead."
         )
+    y_score = check_array(y_score, ensure_2d=False)
+    if y_type == "binary":
+        if y_score.ndim == 2 and y_score.shape[1] != 1:
```

Comment:
The `y_score.shape[1] != 1` here is to support `test_format_invariance_with_1d_vectors`. Based on the test we treat `y_score` with one class as binary.